### ISSUE TITLE
Issue with running PyFLEXTRKR on NEXRAD data

### ISSUE THREAD

**Issue by oluwafemiomitusa** [FlexTRKR/PyFLEXTRKR#80]
I am trying to track convective cells in two NEXRAD datasets but I'm having issues deciphering what the error might be. I gridded the data using the grid_radar_pyart.py in the /pyflxtrkr folder.

This is the error I am getting:

```
Running PyFLEXTRKR ...
2024-01-13 12:33:20,983 - __main__ - INFO - Running in serial.
2024-01-13 12:33:20,983 - pyflextrkr.idfeature_driver - INFO - Identifying features from raw data
2024-01-13 12:33:20,989 - pyflextrkr.idfeature_driver - INFO - Total number of files to process: 2
2024-01-13 12:33:31,715 - pyflextrkr.idcells_reflectivity - INFO - /home/omitusaf/PyFLEXTRKR/Data_results//tracking/cloudid_20210628_000535.nc
2024-01-13 12:33:40,030 - pyflextrkr.idcells_reflectivity - INFO - /home/omitusaf/PyFLEXTRKR/Data_results//tracking/cloudid_20210628_001054.nc
2024-01-13 12:33:40,033 - pyflextrkr.idfeature_driver - INFO - Done with features from raw data.
2024-01-13 12:33:40,033 - __main__ - INFO - Calculating domain mean advection.
2024-01-13 12:33:40,034 - pyflextrkr.advection_tiles - INFO - Found 2 files.
Traceback (most recent call last):
  File "/home/omitusaf/PyFLEXTRKR/config/../runscripts/run_celltracking.py", line 49, in <module>
    driftfile = calc_mean_advection(config)
  File "/home/omitusaf/PyFLEXTRKR/pyflextrkr/advection_tiles.py", line 277, in calc_mean_advection
    x_y = movement_of_storm_fft_l(
  File "/home/omitusaf/PyFLEXTRKR/pyflextrkr/advection_tiles.py", line 213, in movement_of_storm_fft_l
    y1, x1 = movement_of_storm_fft(
  File "/home/omitusaf/PyFLEXTRKR/pyflextrkr/advection_tiles.py", line 159, in movement_of_storm_fft
    y, x = -1 * phase_cross_correlation(
ValueError: not enough values to unpack (expected 2, got 0)
Tracking is done.
```

I have also included my config file:
---
```
# Example radar cell tracking configuration file (500m resolution)

# Identify features to track
run_idfeature: True
# Tracking steps to run
run_advection: True
# Track single consecutive feature files
run_tracksingle: True
# Run tracking for all files
run_gettracks: True
# Calculate feature statistics
run_trackstats: True
# Map tracking to pixel files
run_mapfeature: True

# Parallel processing set up
# run_parallel: 1 (local cluster), 2 (Dask MPI)
run_parallel: 0
nprocesses: 4  # Number of processors to use if run_parallel=1
dask_tmp_dir: '/tmp'  # Dask temporary directory if run_parallel=1
timeout: 360  # [seconds] Dask timeout limit

# Start/end date and time
startdate: '20210628.0000'
enddate: '20210628.2359'

# Specify tracking input data date/time string format
# This is the preprocessed file that contains Tb & rainrate
# E.g., databasename20181101.011503.nc --> yyyymodd.hhmmss
# E.g., databasename2018-11-01_01:15:00 --> yyyy-mo-dd_hh:mm:ss
time_format: 'yyyymodd.hhmmss'
databasename: 'KHGX'

# Input files directory
clouddata_path: '/home/omitusaf/PyFLEXTRKR/Data_results/input/'
# Working directory for the tracking data
root_path: '/home/omitusaf/PyFLEXTRKR/Data_results/'
# Working sub-directory names
tracking_path_name: 'tracking'
stats_path_name: 'stats'
pixel_path_name: 'celltracking'

# Terrain file
# terrain_file: 'INPUT_DIR/Terrain_RangeMask.nc'
# Elevation variable to use in terrain file
# elev_varname: 'hgt'
# Range mask variable to use in terrain file
# rangemask_varname: 'mask110'

# Specify types of feature being tracked
# This adds additional feature-specific statistics to be computed
feature_type: 'radar_cells'

# Advection calculation parameters
advection_field_threshold: 10    # radar reflectivity threshold [dBZ]
advection_med_filt_len: 9         # median filter length for advection speed time seires [number of points]
advection_max_movement_mps: 60    # maximum advection speed allowed [m/s]
advection_mask_method: 'greater'  # masking method
advection_buffer: 30  # number of grid points around the edge of domain to buffer
advection_size_threshold: 10  # number of min valid points to calculate advection
advection_tiles: [1,1]   # number of tiles to calculate advection [y,x]
advection_filename: 'advection_'

# Cell identification parameters
# Parameter in Steiner
# The background reflectivity fall off point in the cosine fucntion,
# if background reflectivity > absConvThres [dBZ], the grid point is a convective core
absConvThres: 60
# Minimum difference to background reflectivity [dB] in the cosine function
minZdiff: 10
# Any grid point reflectivity > truncZconvThres [dBZ] is automatically a convective core
truncZconvThres: 55
# These two reflectivity values sets up the step function background Ze values
# They control the dilation of grids surrounding the convective cores to be included as "convective region"
mindBZuse: 25
dBZforMaxConvRadius: 60
# Convective radius step-function increment [km]
conv_rad_increment: 0.5
# Convective radius step-function start value [km]
conv_rad_start: 1.0
# Background reflectivity step-function increment [dB]
bkg_refl_increment: 5
# Convolution method to calculate background reflectivity: 'ndimage' (default), 'signal' (FFT)
convolve_method: 'ndimage'
# Maximum convective radius dilation [km]
maxConvRadius: 5
# Define a set of radii for final step of convective cell expansion [km]
# This is for the purpose of making the convective region larger for tracking convective cells
radii_expand: [1,2,3,4,5]
# Weak echo threshold [dBZ]
weakEchoThres: 15
# Background radius to calculate background reflectivity intensity [km]
bkgrndRadius: 11
# Minimum area threshold to be a convective core [km^2]
min_corearea: 4
# Number of vertical level gaps allowed in calculating echo-top height
echotop_gap: 4
# Minimum height above surface elevation to filter the data
sfc_dz_min: 1000
# Maximum height above surface [m] to calculate low-level max reflectivity
# (similar to lowest-level composite reflectiivty)
sfc_dz_max: 3000
# Set equivalent radar sensitivity [dBZ]
radar_sensitivity: 0.0
# Flag to output additional Steiner variables for diagnostic purpose
return_diag: True
# Radar file data dimensions and reflectivity variable name
dx: 500  # meter
dy: 500  # meter
x_dimname: 'x'
y_dimname: 'y'
z_dimname: 'z'
time_dimname: 'time'
time_coordname: 'time'
x_varname: 'x'
y_varname: 'y'
z_varname: 'z'
lon_varname: 'point_longitude'
lat_varname: 'point_latitude'
reflectivity_varname: 'reflectivity'
# Input data source: 'radar', 'wrf', 'wrf_regrid'
input_source: 'radar'

# Specify data structure
datatimeresolution: 0.0833     # hours
pixel_radius: 0.5      # km
ref_varname: 'dbz_comp'

# Tracking parameters
timegap: 0.25           # hour
othresh: 0.3           # overlap percentage threshold
maxnclouds: 1000       # Maximum number of clouds in one snapshot
nmaxlinks: 10          # Maximum number of clouds that any single cloud can be linked to
duration_range: [2, 100]   # A vector [minlength,maxlength] to specify the duration range for the tracks
# Flag to remove short-lived tracks [< min(duration_range)] that are not mergers/splits with other tracks
# 0:keep all tracks; 1:remove short tracks
remove_shorttracks: 1
# Set this flag to 1 to write a dense (2D) trackstats netCDF file
# Note that for datasets with lots of tracks, the memory consumption could be very large
trackstats_dense_netcdf: 1
# Minimum time difference threshold to match track stats with cloudid pixel files
match_pixel_dt_thresh: 60.0  # seconds

# Define tracked feature variable names
feature_varname: 'feature_number'
nfeature_varname: 'nfeatures'
featuresize_varname: 'npix_feature'

# Track statistics output file dimension names
tracks_dimname: 'tracks'
times_dimname: 'times'
fillval: -9999
pixeltracking_filebase: 'celltracks_'

```

**Comment by feng045** [comment]
This step where you crashed is for calculating the cell advection speed in the domain, using the [SciKit-image phase_cross_correlation](https://scikit-image.org/docs/stable/api/skimage.registration.html#skimage.registration.phase_cross_correlation) function.

It looks like it wasn't returning the two advection values (y, x) as expected for some reason.  I'm not sure what your own NEXRAD radar looks like, did you try running the [demo NEXRAD script](https://github.com/FlexTRKR/PyFLEXTRKR/blob/main/config/demo_cell_nexrad.sh) with the provided example data to see if it runs through?  If the demo data runs through, then perhaps it's related to the data you provided.

**Comment by oluwafemiomitusa** [comment]
Yes, I have run the original script but the error persists.

**Comment by lyuhm** [comment]
I have the same error. Have you solved this problem? Can you share it with me?

**Comment by feng045** [comment]
Please pull the latest version.  This issue has been fixed in #86. Specifically, in [this commit](https://github.com/FlexTRKR/PyFLEXTRKR/pull/86/commits/8b797384b5f88ffd8733b2c867a114f2a1908695).

**Comment by lyuhm** [comment]
Thanks, the error has been fixed
